esphome/components/zyaura/zyaura.cpp

```diff
@@ -57,38 +57,46 @@ void IRAM_ATTR ZaSensorStore::interrupt(ZaSensorStore *arg) {
 void IRAM_ATTR ZaSensorStore::set_data_(ZaMessage *message) {
   switch (message->type) {
     case HUMIDITY:
-      this->humidity = (message->value > 10000) ? NAN : (message->value / 100.0f);
+      this->humidity = message->value;
       break;
-
     case TEMPERATURE:
-      this->temperature = (message->value > 5970) ? NAN : (message->value / 16.0f - 273.15f);
+      this->temperature = message->value;
       break;
-
     case CO2:
-      this->co2 = (message->value > 10000) ? NAN : message->value;
-      break;
-
-    default:
+      this->co2 = message->value;
       break;
   }
 }
 
-bool ZyAuraSensor::publish_state_(sensor::Sensor *sensor, float *value) {
-  // Sensor doesn't added to configuration
+bool ZyAuraSensor::publish_state_(ZaDataType data_type, sensor::Sensor *sensor, uint16_t *data_value) {
+  // Sensor wasn't added to configuration
   if (sensor == nullptr) {
     return true;
   }
 
-  sensor->publish_state(*value);
+  float value;
+  switch (data_type) {
+    case HUMIDITY:
+      value = (*data_value > 10000) ? NAN : (*data_value / 100.0f);
+      break;
+    case TEMPERATURE:
+      value = (*data_value > 5970) ? NAN : (*data_value / 16.0f - 273.15f);
+      break;
+    case CO2:
+      value = (*data_value > 10000) ? NAN : *data_value;
+      break;
```

Comment:
Curious that compile warning didn't show up on CI...
```suggestion
      break;
    default:
      return false;
```

Reply:
I compiled the new code using `pio run --target upload` from within the project folder after compiling it once with `esphome compile`, because i was not able to compile it as a local external component/the git issue syntax, e.g. `source: github://pr#2751`.

This is my first encounter with ESPHome, so I'm sure that I messed something up during configuring the external component, but somewhere in the docs I read about the possibility to compile it with pio run. Maybe this is doing different things than esphome compile?

Reply:
> Maybe this is doing different things than esphome compile?

No, you're doing it right, I get the warning too. CI actually [gets it as well](https://github.com/esphome/esphome/runs/4245943944?check_suite_focus=true#step:11:473), but I think the matcher is broken.

Reply:
It shouldn't need that because all cases are covered.  I'll just initialize the variable...

Reply:
> It shouldn't need that because all cases are covered.

The problem is that in C++ enumerations are really just an integer, so an enum variable can have values that aren't part of the enum.